qiskit/transpiler/runningpassmanager.py

```diff
@@ -311,42 +322,76 @@ class FlowControllerLinear(FlowController):
     def __init__(self, passes, options):  # pylint: disable=super-init-not-called
         self.passes = self._passes = passes
         self.options = options
+        self.property_set = None
+
+    def do_passes(self, pass_manager, dag, property_set):
+        """ In the context of pass_manager, runs the pass on the dag
+        Args:
+            pass_manager (PassManager): A PassManager object.
+            dag (DAGCircuit): The dag on which the pass is ran.
+            property_set (PropertySet): It will be use for parametrizing the flow controller
+        Returns:
+            DAGCircuit: The dag after the pass.
+        """
+        self.property_set = property_set
+        for pass_ in self:
+            dag = pass_manager._do_pass(pass_, dag)
+        return dag
 
 
 class DoWhileController(FlowController):
     """Implements a set of passes in a do-while loop."""
 
-    def __init__(self, passes, options, do_while=None,
-                 **partial_controller):
+    def __init__(self, passes, options, do_while=None, **flow_controller):
         self.do_while = do_while
         self.max_iteration = options['max_iteration']
-        super().__init__(passes, options, **partial_controller)
+        self.property_set = None
+        super().__init__(passes, options, **flow_controller)
 
     def __iter__(self):
         for _ in range(self.max_iteration):
+            self.passes.property_set = self.property_set
             for pass_ in self.passes:
                 yield pass_
 
-            if not self.do_while():
+            if not self.do_while(self.property_set):
                 return
-
         raise TranspilerError("Maximum iteration reached. max_iteration=%i" % self.max_iteration)
 
 
 class ConditionalController(FlowController):
     """Implements a set of passes under a certain condition."""
 
-    def __init__(self, passes, options, condition=None,
-                 **partial_controller):
+    def __init__(self, passes, options, condition=None, **flow_controller):
         self.condition = condition
-        super().__init__(passes, options, **partial_controller)
+        super().__init__(passes, options, **flow_controller)
 
     def __iter__(self):
-        if self.condition():
+        if self.condition(self.property_set):
+            self.passes.property_set = self.property_set
             for pass_ in self.passes:
                 yield pass_
 
 
+class RollbackIfController(FlowController):
+    """ The set of passes is rollbacked if the condition condition is true."""
+
+    def __init__(self, passes, options, rollback_if=None, **_):
+        self.rollback_if = rollback_if
+        super().__init__(passes, options)
+
+    def do_passes(self, pass_manager, dag, property_set):
+        original_property_set = deepcopy(pass_manager.property_set)
+        dag_copy = deepcopy(dag)
+        for pass_ in self:
+            dag = pass_manager._do_pass(pass_, dag_copy)
```

Comment:
this would be a bug because all the passes are applied on the same input.
```suggestion
            dag = pass_manager._do_pass(pass_, dag)
```

Reply:
good catch. Fixed in 9f59e641d